doc/Architecture-overview/index.md

```diff
@@ -2,7 +2,158 @@
+
+## Active load balancing
+
+Kaa SDK choose Bootstrap and Operations service instances pseudo-randomly during session initiation.
+However, under the conditions when the cluster is heavily loaded, random endpoints distribution may not be good enough.
+Further, as a node joins the cluster, there is a need to re-balance the load across the new cluster set up.
+Kaa server uses the active load balancing approach to instruct some of the endpoints to reconnect to a different Operations service, thus equalizing the load across the nodes.
+The algorithm takes servers load information (connected endpoints count, load average, etc.) published by Kaa nodes as an input, and periodically recalculates weights of each node.
+Further, the overloaded nodes are instructed to redirect some of the connecting endpoints to a different node.
+
+A similar approach can be used for offloading all of the load from a node subject to a scheduled service, or to gradually migrate the cluster across the physical or virtual machines.
+Doing so requires setting of a custom load balancing strategy.
+
+# Endpoint SDK
+
+The Kaa endpoint SDK is a library which provides communication, data marshalling, persistence, and other functions available in Kaa for specific type of an endpoint (e.g. [C-based](Using-Kaa-C-endpoint-SDK), [C++-based](Using-Kaa-Cpp-endpoint-SDK), [Java-based](Using-Kaa-Java-endpoint-SDK), [Android-based](), [Objective-C-based](Using-Kaa-Objective-C-endpoint-SDK)).
+The client SDK abstracts the communication protocol, data persistence, and other implementation details that may be specific for any concrete solution based on Kaa.
+
+Endpoint SDK helps to save time on development routine and allows to concentrate on your business logic.
+
+
+# Kaa instance
+
+<img src="attach/logical-concepts.png" width="1382" height="703" style="display: block;margin-left: auto;margin-right: auto;">
+
+*Kaa instance* (interchangeable with *Kaa deployment*) is a particular installation of the Kaa platform, either as a [single node]({{root_url}}Administration-guide/System-installation/Single-node-installation/), or a [clustered deployment]({{root_url}}Administration-guide/System-installation/Cluster-setup/).
+
+An *application* in Kaa defines the set of data models, the corresponding flows among the endpoints and Kaa server, and processing rules.
+Kaa applications are agnostic to the specific target platform, operating system, or the client software implementation.
+For example, two firmware implementations for a pressure sensor will differ between Arduino and STM32 platforms, yet will be considered the same application in Kaa, as long as they report identically structured telemetry data.
+
+The Kaa platform is *multi-tenant*, and may support multiple independent business entities out of a single instance.
+As illustrated in the following diagram, applications belong to tenants, while endpoints register within applications.
+
```

Comment:
As illustrated in the following diagram - there is no following diagram here. It is "preceding". Also, we can move the diagram here